pkg/master/controller/crdregistration/crdregistration_controller.go

```diff
@@ -208,19 +223,50 @@ func (c *crdRegistrationController) handleVersionUpdate(groupVersion schema.Grou
 				continue
 			}
 
-			c.apiServiceRegistration.AddAPIServiceToSync(&apiregistration.APIService{
+			apiService := &apiregistration.APIService{
 				ObjectMeta: metav1.ObjectMeta{Name: apiServiceName},
 				Spec: apiregistration.APIServiceSpec{
 					Group:                groupVersion.Group,
 					Version:              groupVersion.Version,
 					GroupPriorityMinimum: 1000, // CRDs should have relatively low priority
 					VersionPriority:      100,  // CRDs will be sorted by kube-like versions like any other APIService with the same VersionPriority
 				},
-			})
+			}
+			c.apiServiceRegistration.AddAPIServiceToSync(apiService)
+			// Serve CRD OpenAPI spec if we have non-nil openAPIAggregationManager
+			if c.openAPIAggregationManager != nil {
```

Comment:
Can we move all of this into apiextensions-apiserver? In other words: only use the informer as a trigger that one of the delegated apiservers (which in this case happens to be apiextensions-apiserver) has updated resources and then pull the new openapi spec fromt the delegated apiserver to merge it in?